tests/models/test_structures.py

```diff
@@ -38,4 +38,41 @@ def test_bad_structures(bad_structures, mapper):
         # This is for helping devs finding any errors that may occur
         print(f"Trying structure number {index} from 'test_bad_structures.json'")
         with pytest.raises(ValidationError):
-            StructureResource(**mapper(MAPPER).map_back(structure))
+            try:
+                StructureResource(**mapper(MAPPER).map_back(structure))
+            except Exception as exc:
+                print(exc)
+                raise exc
+
+
+deformities = (
+    None,
+    {"chemical_formula_anonymous": "AaBC"},
+    {"chemical_formula_anonymous": "BC"},
+    {"chemical_formula_anonymous": "A1B1"},
+    {"chemical_formula_anonymous": "BC1"},
+    {"chemical_formula_anonymous": "A9C"},
+    {"chemical_formula_hill": "SiGe"},
+    {"chemical_formula_hill": "GeHSi"},
+    {"chemical_formula_reduced": "Ge1Si"},
+    {"chemical_formula_reduced": "GeSi1"},
+    {"chemical_formula_reduced": "SiGe2"},
+    {"chemical_formula_reduced": "abcd"},
+    {"chemical_formula_reduced": "a2BeH"},
+    {"chemical_formula_reduced": "............"},
+    {"chemical_formula_reduced": "Ag6 Cl2"},
+)
+
+
+@pytest.mark.parametrize("deformity", deformities)
+def test_structure_deformities(good_structure, deformity):
+    """Make specific checks upon performing single invalidating deformations
+    of the data of a good structure.
+
+    """
+    if deformity is None:
+        StructureResource(**good_structure)
+    else:
+        good_structure["attributes"].update(deformity)
+        with pytest.raises(ValidationError):
```

Comment:
You should add `match=""` here to look for the field in question expected to be the culprit indeed also is the culprit. E.g.:
```suggestion
        with pytest.raises(ValidationError, match=fr".*{deformity.keys()[0]}.*"):
```

Reply:
I tried this, but ran into exactly the same problem we had in another PR where the match does not work through pytest (but does with `re`). Still can't figure out what's going wrong.

Reply:
Huh, can't find the old discussion we had about this, and this does indeed seem to work, I wonder if pytest 6 fixed something

Reply:
Or we just didn't add the `.*` in the string?

Reply:
Perhaps, I think maybe I was trying to match the entire message last time